### PR TITLE
docs: doc about static style extract

### LINKED ISSUE #39891: Next.js fails to render style tags when server emits HTML
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/eager-shirley-k0zlcq?file=%2FREADME.md)

### Steps to reproduce

1. Create a new Next.js 13 app with the "app" directory feature:
```
npx create-next-app@latest --experimental-app
# or
yarn create next-app --experimental-app
# or
pnpm create next-app --experimental-app
```

2. Add ant design to the project
3. Add a few random components in the "app/page.tsx" file. Don't forget to add "use client" at the top of the file
4. Run the "dev" script and open the browser at the specified port
5. Observe how the page loads without styles for a brief second. After a few moments, the styles get downloaded.

I noticed this issue was introduced with @ant-design/cssinjs v1.3.1. Here is a [reproduction link](https://codesandbox.io/p/sandbox/purple-sun-8o8z8e?file=%2FREADME.md) where I modified the yarn.lock file to load an older version on cssinjs (v.1.1.0) where everything loads ok.

Here are also some videos to see how it looks like.  Pay attention at the reload button to see when I refresh the page:

Before (using a yarn.lock file that requires cssinjss v.1.1.0):

https://user-images.githubusercontent.com/4083652/209926786-2127bca0-6708-44d0-b962-166807bcc718.mov

After (using the latest version of AntDesign which comes with cssinjs v.1.3.1):

https://user-images.githubusercontent.com/4083652/209926824-ecdb3f27-f805-48fa-97c4-72b6e6dfe468.mov

Other mentions of the issue:
- https://github.com/ant-design/ant-design/issues/39885
- https://github.com/ant-design/cssinjs/issues/69#issuecomment-1367104002

### What is expected?

The styles to be loaded together with the HTML as before

### What is actually happening?

The styles are loaded after the page loads the HTML

| Environment | Info |
| --- | --- |
| antd | 5.1.1 |
| React | 18.2.0 |
| System | MacOS Ventura 13.0.1 |
| Browser | Chrome 108 |

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #39891

### 💡 Background and solution
doc about static style extract

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | doc about static style extract |
| 🇨🇳 Chinese | 关于静态样式导出的相关文章 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

File: docs/react/customize-theme.zh-CN.md [hunk 1]
```diff
@@ -240,24 +240,18 @@ const theme = {
 
 ### 服务端渲染
 
-使用 `@ant-design/cssinjs` 将所需样式抽离：
+由于`ant-design`从 `5.0`起全面使用`css-in-js`的方式替代了`less`，但现有方案在`SSR`场景下只能将相关样式直接写入`HTML`的行间样式当中，导致`HTML`文件异常庞大，相关问题讨论详见：[#39891](https://github.com/ant-design/ant-design/issues/39891)，影响首屏渲染的速度。基于上述目的，我们提供了`@ant-design/static-style-extract`支持全量组件样式抽离（交互组件等非 SSR 场景显示的组件样式除外，如`Modal`，具体的黑名单列表详见：[static-style-extract](https://github.com/ant-design/static-style-extract/blob/610aae06c609ed366525d92199b8c56553a1e08f/src/index.tsx#L10)）。使用 `@ant-design/static-style-extract` 将得到我们预期的一个`css`字符串，在项目中，我们可以通过自己的方法将这个样式字符串写入到文件中引用。
```
Comment (zombieJ) [inline docs/react/customize-theme.zh-CN.md:243]: 其实也不是只能。只是不是所有用户知道怎么封装而已，antd v5 官网发布之初就是 extract css 的。

```markdown
服务端渲染样式有两种方案，它们各有优缺点：
* 整体导出：提前烘焙 antd 组件样式为 css 文件，在页面中时引入。好处是打开任意页面时如传统 css 方案一样都会复用同一套 css 文件以命中缓存，缺点是如果页面中存在多主题，则需要额外进行烘焙
* 内联方式：在渲染时 balabala，好处是 xxx，缺点则是会使得 HTML balabala，相关讨论参考 balabala
```

Comment (zombieJ) [reply docs/react/customize-theme.zh-CN.md:243]: 然后把这个 `@ant-design/static-style-extract` 介绍挪到下面去，这里只做简单介绍，不用把方案细则说出来。

File: docs/react/customize-theme.zh-CN.md [hunk 2]
```diff
@@ -240,24 +240,18 @@ const theme = {
...
 
-```tsx
-import { createCache, extractStyle, StyleProvider } from '@ant-design/cssinjs';
+#### 直接注入行间样式
```
Comment (zombieJ) [inline docs/react/customize-theme.zh-CN.md:245]: 这个直接的注入还是原来的代码和介绍，不应该用到 `@ant-design/static-style-extract`

File: docs/react/customize-theme.zh-CN.md (hunk 1 above)
Comment (kiner-tang) [reply docs/react/customize-theme.zh-CN.md:243]: 好的

File: docs/react/customize-theme.zh-CN.md (hunk 2 above)
Comment (kiner-tang) [reply docs/react/customize-theme.zh-CN.md:245]: 好

File: docs/react/customize-theme.zh-CN.md (hunk 1 above)
Comment (kiner-tang) [reply docs/react/customize-theme.zh-CN.md]: done

File: docs/react/customize-theme.zh-CN.md [hunk 3]
```diff
@@ -274,6 +281,60 @@ export default () => {
...
+
+const css = extractStyle();
+fs.writeFileSync(outputPath, css);
+
+console.log(`🎉 Antd CSS generated at ${outputPath}`);
+```
+
+你可以选择在启动开发命令或编译前执行这个脚本，运行上述脚本将会在当前项目的指定（如：`public`目录）目录下直接生成一个全量的`antd.min.css`文件。
+
+以`Next.js`为例（[参考示例](https://github.com/ant-design/create-next-app-antd)）：
```
Comment (zombieJ) [inline docs/react/customize-theme.zh-CN.md:303]: > 以 `Next.js` 为例

英文前后加空格~

File: docs/react/customize-theme.zh-CN.md [hunk 4]
```diff
@@ -274,6 +281,60 @@ export default () => {
...
+import '../styles/globals.css'; // 添加这行
+
+export default function App({ Component, pageProps }: AppProps) {
+  return (
+    <StyleProvider hashPriority="high">
+      <Component {...pageProps} />
+    </StyleProvider>
+  );
+}
+```
```
Comment (zombieJ) [inline docs/react/customize-theme.zh-CN.md:334]: 这里再加个小小标题，然后介绍一下混合主题如何烘焙

File: docs/react/customize-theme.zh-CN.md (hunk 3 above)
Comment (afc163) [reply docs/react/customize-theme.zh-CN.md:303]: prettier 看来没生效

Comment (afc163) [reply docs/react/customize-theme.zh-CN.md:303]: Next.js 不是代码，不用加 `

File: docs/react/customize-theme.zh-CN.md [hunk 5]
```diff
@@ -274,6 +281,60 @@ export default () => {
...
+// scripts/genAntdCss.mjs
+import fs from 'fs';
+import { extractStyle } from '@ant-design/static-style-extract';
+
+const outputPath = './public/antd.min.css';
+
+const css = extractStyle();
+fs.writeFileSync(outputPath, css);
+
+console.log(`🎉 Antd CSS generated at ${outputPath}`);
```
[the remark this answers is not in the document]
Comment (kiner-tang) [reply docs/react/customize-theme.zh-CN.md]: done

File: docs/react/customize-theme.zh-CN.md (hunk 3 above)
Comment (kiner-tang) [reply docs/react/customize-theme.zh-CN.md]: done

File: docs/react/customize-theme.zh-CN.md (hunk 4 above)
Comment (kiner-tang) [reply docs/react/customize-theme.zh-CN.md:332]: ok

File: docs/react/customize-theme.zh-CN.md [hunk 6]
```diff
@@ -274,6 +281,108 @@ export default () => {
...
+
+#### 混合主题
+
+如果你的项目中使用了混合主题，可以尝试通过以下方式进行烘焙：
+
+```tsx
+import { extractStyle } from '@ant-design/static-style-extract';
+import { ConfigProvider } from 'antd';
+
+const cssText = extractStyle((node) => (
```
Comment (zombieJ) [inline docs/react/customize-theme.zh-CN.md:363]: 这里再补一下外部的：
```tsx
const cssText = extractStyle((node) => (<>
  {/* .... */}
  <CP theme={...}>{node}</CP>
  {/* .... */}
  <CP theme={...}><CP theme={...}>{node}</CP></CP>
</>);
```

Comment (kiner-tang) [reply docs/react/customize-theme.zh-CN.md:363]: done

**Review by zombieJ — APPROVED** [review]

**Pull request merged**